Observer/src/observer/Channel.java

```diff
@@ -0,0 +1,31 @@
+package observer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Channel {
+    private List<Subscriber> subscribers = new ArrayList<>();
+    private String title;
+
+    public void subscribeChannel(Subscriber subscriber) {
+        subscribers.add(subscriber);
+        System.out.println(subscriber.getSubscriberName() + " has subscribed your channel");
+    }
+
+    public void unSubscribeChannel(Subscriber subscriber) {
+        subscribers.remove(subscriber);
+        System.out.println(subscriber.getSubscriberName() + " has unsubscribed your channel");
+    }
+
+    public void notifySubscribers(){
+        for(Subscriber subscriber: subscribers){
+            subscriber.update();
```

Comment:
channel is not notifying any info to the subscriber here

Reply:
It's the channel which notifies the video upload status. For the scope of this problem statement, It's the responsibility of the channel to notify the video status to the subscribers